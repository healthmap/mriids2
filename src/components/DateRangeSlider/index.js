import React, { useState } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { changeDateRange } from "../../actions/filters";
import { Slider } from "@material-ui/core";
import { DateRangeSliderContainer } from "../styled-components/DateRangeSliderContainer";
import { ebolaInitialDateRange } from "../../constants/DateRanges";

const DateRangeSlider = (props) => {
  const [numberOfWeeks, setNumberOfWeeks] = useState([0, 68]);

  const handleRangeChange = (event, newRangeArray) => {
    // Only execute this block if the newRangeArray is different from the numberOfWeeks state.
    if (numberOfWeeks !== newRangeArray) {
      // Here we are getting the new dateRange.from date value for the filters
      let newFromDate = new Date(ebolaInitialDateRange.dateRange.from);
      const fromDateChange = 7 * newRangeArray[0];
      newFromDate.setDate(newFromDate.getDate() + fromDateChange);
      // Here we are getting the new dateRange.to date value for the filters
      let newToDate = new Date(ebolaInitialDateRange.dateRange.to);
      const toDateChange = 7 * (newRangeArray[1] - 68);
      newToDate.setDate(newToDate.getDate() + toDateChange);

      props.changeDateRange([newFromDate, newToDate]);
      setNumberOfWeeks(newRangeArray);
    }
  };

  return (
    <DateRangeSliderContainer>
      <Slider value={numberOfWeeks} onChange={handleRangeChange} />
    </DateRangeSliderContainer>
  );
};

const mapStateToProps = (state) => ({
  filters: state.filters,
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      changeDateRange,
    },
    dispatch
  );

export default connect(mapStateToProps, mapDispatchToProps)(DateRangeSlider);
