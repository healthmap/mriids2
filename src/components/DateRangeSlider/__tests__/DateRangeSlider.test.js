import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import renderer from "react-test-renderer";
import "jest-styled-components";
import DateRangeSlider from "../index";

const mockStore = configureStore([thunk]);

describe("Tests for the connected DateRangeSlider", () => {
  let store;
  let component;

  const initialState = {
    filters: {
      country: "All",
      outbreak: "Ebola Outbreak",
      view: "snapshot",
      projection: false,
      dateRange: { from: new Date(2014, 9, 1), to: new Date(2016, 1, 20) },
    },
    ebola: {
      ebolaData: {
        isFetching: 0,
        data: {},
        error: {},
      },
      ebolaDataCombined: {
        isFetching: 0,
        data: {},
        error: {},
      },
      riskData: {
        isFetching: 0,
        data: {},
        error: {},
      },
    },
  };

  beforeEach(() => {
    store = mockStore(initialState);

    component = renderer.create(
      <Provider store={store}>
        <DateRangeSlider />
      </Provider>
    );
  });
  test("should render with given state from Redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
