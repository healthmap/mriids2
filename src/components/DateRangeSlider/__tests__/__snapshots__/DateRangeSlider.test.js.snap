// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the connected DateRangeSlider should render with given state from Redux store 1`] = `
.c0 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  height: 8rem;
  overflow: hidden;
}

<div
  className="c0"
>
  <span
    className="MuiSlider-root MuiSlider-colorPrimary"
    onMouseDown={[Function]}
  >
    <span
      className="MuiSlider-rail"
    />
    <span
      className="MuiSlider-track"
      style={
        Object {
          "left": "0%",
          "width": "100%",
        }
      }
    />
    <input
      type="hidden"
      value="0,68"
    />
    <span
      aria-orientation="horizontal"
      aria-valuemax={68}
      aria-valuemin={0}
      aria-valuenow={0}
      className="MuiSlider-thumb MuiSlider-thumbColorPrimary"
      data-index={0}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      role="slider"
      style={
        Object {
          "left": "0%",
        }
      }
      tabIndex={0}
    />
    <span
      aria-orientation="horizontal"
      aria-valuemax={68}
      aria-valuemin={0}
      aria-valuenow={68}
      className="MuiSlider-thumb MuiSlider-thumbColorPrimary"
      data-index={1}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      role="slider"
      style={
        Object {
          "left": "100%",
        }
      }
      tabIndex={0}
    />
  </span>
</div>
`;
